security/oc_tls.c

```diff
@@ -393,18 +393,32 @@ process_drop_event_for_removed_endpoint(oc_process_event_t ev,
 {
   const oc_endpoint_t *endpoint = (const oc_endpoint_t *)user_data;
   if (ev != oc_event_to_oc_process_event(RI_TO_TLS_EVENT) &&
-      ev != oc_event_to_oc_process_event(UDP_TO_TLS_EVENT)) {
+      ev != oc_event_to_oc_process_event(UDP_TO_TLS_EVENT) &&
+      ev != oc_event_to_oc_process_event(OUTBOUND_NETWORK_EVENT) &&
+      ev != oc_event_to_oc_process_event(INBOUND_NETWORK_EVENT)) {
     return false;
   }
   oc_message_t *message = (oc_message_t *)data;
   if (oc_endpoint_compare(&message->endpoint, endpoint) == 0) {
 #if OC_DBG_IS_ENABLED
-    oc_string64_t endpoint_str;
-    oc_endpoint_to_string64(&message->endpoint, &endpoint_str);
-    OC_DBG("oc_tls: dropping %s message for removed endpoint(%s)",
-           (ev == oc_event_to_oc_process_event(RI_TO_TLS_EVENT)) ? "sent"
-                                                                 : "received",
+    oc_string_t endpoint_str;
+    oc_endpoint_to_string(&message->endpoint, &endpoint_str);
```

Comment:
oc_endpoint_to_string64 to avoid allocation and deallocation

Reply:
fixed